src/NuGet.Clients/NuGet.VisualStudio.Implementation/SolutionExplorer/Models/AssetsFileDependenciesSnapshot.cs

```diff
@@ -196,6 +207,77 @@ static ImmutableArray<AssetsFileLogMessage> ParseLogMessages(LockFile lockFile,
             }
 
             return builder.ToImmutable();
+
+            Dictionary<string, LogLevel> BuildLevelByLibrary()
```

Comment:
Can we get some tests for this logic?

Reply:
I looked at testing but wasn't sure how I could do so in the time I carved out for this. Any pointers on how I could spin one up without a lot of boilerplate?

Reply:
We have some test cases that just add a sample project.assets.json resource, and just use that as a predetermined project setup. https://github.com/NuGet/NuGet.Client/blob/dev/test/NuGet.Core.Tests/NuGet.Build.Tasks.Pack.Test/compiler/resources/project.assets.json

There are ways to create your own within the test as well: 

https://github.com/NuGet/NuGet.Client/blob/6b9e4481b9c23665ceb3192b9964921bcbb67c30/test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommandTests/RestoreCommandTests.cs#L3220-L3235

The project declaration can also be read like: 

https://github.com/NuGet/NuGet.Client/blob/6b9e4481b9c23665ceb3192b9964921bcbb67c30/test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommandTests/RestoreCommandTests.cs#L1847-L1871

The test packages have dependencies elements so you can create a graph as you see fit.

Reply:
To test this I needed a unique assets file. I brought one in to the unit test directly.

Tests now added.